app/services/badge/bestowment_service.rb

```diff
@@ -0,0 +1,55 @@
+# Handles creating and updating bestowments for a (badge, user) pair.
+class Badge
+  class BestowmentService
+    attr_reader :badge, :user
+
+    # @param [Class<Badge>] the badge to handle bestowments for
+    # @param [User] the user to bestow this badge upon
+    def initialize(badge, user)
+      @badge = badge
```

Comment:
Does ```@badge``` need to be initialized? 
I'm not able to get anything to return when ```badge.progress_for(user)``` is called from ```ranks```